### PR TITLE
Add file download

### PR DESCRIPTION
**Pull request by thompsond**
- Add base code for downloading images
- Format and remove unused imports

### CODE REVIEW

File: src/main/java/com/google/step/Authenticate.java [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
 
-/***
-    This servlet handles the first step of the authentication process with OAuth.
-    When the user visits the home page they are immediately sent here to begin the
-    authentication and redirection process.
-***/
+/**
+ * * This servlet handles the first step of the authentication process with OAuth. When the user
+ * visits the home page they are immediately sent here to begin the authentication and redirection
+ * process. *
```
Comment (gabbygermanson) [inline src/main/java/com/google/step/Authenticate.java:14]: bring the end star to next line? Same in backenddatastore, formlocations, and more files if so

File: src/main/java/com/google/step/CorsContextListener.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import com.google.common.collect.ImmutableList;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.annotation.WebListener;
+
+/**
+ * ****** This servlet is a context listenter that runs at application startup. It configures the
+ * Cloud Storage bucket to accept CORS requests from the project URL. ******
```
Comment (gabbygermanson) [inline src/main/java/com/google/step/CorsContextListener.java:15]: This many stars needed?

File: src/main/java/com/google/step/FrontendQueryDatastore.java [hunk 3]
```diff
@@ -1,63 +1,61 @@
 package com.google.step;
 
+import static java.lang.Math.toIntExact;
+
```
Comment (gabbygermanson) [inline src/main/java/com/google/step/FrontendQueryDatastore.java:4]: remove line

File: src/main/java/com/google/step/FrontendQueryDatastore.java [hunk 4]
```diff
@@ -190,24 +190,31 @@ private Filter buildDateFilters(int monthFrom, int monthTo, int yearFrom, int ye
...
-        Converts each individual entity into a MapImage object.
-    ***/
-    private MapImage entityToMapImage(Entity entity) { 
+    /*
+     *   NOTE: entity.get"Type" (i.e. entity.getDouble) will return either DatastoreException
+     *   if the property doesn't exist, or a ClassCastException if the value is the wrong type
+     */
+    /**
+     * * Helper function for entitiesToMapImages. Converts each individual entity into a MapImage
+     * object. *
```
Comment (gabbygermanson) [inline src/main/java/com/google/step/FrontendQueryDatastore.java:199]: Combine documentation with above

File: src/main/java/com/google/step/MapImage.java [hunk 5]
```diff
@@ -102,7 +107,7 @@ public int getMonth() {
 
     public int getYear() {
         return year;
-    }  
+    }
```
Comment (gabbygermanson) [inline src/main/java/com/google/step/MapImage.java:111]: extra line

File: src/main/java/com/google/step/CorsContextListener.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    @Override
+    public void contextInitialized(ServletContextEvent sce) {
+        Storage storage = StorageOptions.getDefaultInstance().getService();
+        Bucket bucket = storage.get(BUCKET_NAME);
+        String origin = "https://map-snapshot-step.uc.r.appspot.com";
+        HttpMethod method = HttpMethod.GET;
+        String responseHeader = "Content-Type";
+        int maxAgeSeconds = 3600;
+
+        Cors cors =
```
Comment (cgregori) [inline src/main/java/com/google/step/CorsContextListener.java:31]: nit: tab spacing here. Maybe: 

```suggestion
Cors cors = Cors.newBuilder()
    .setOrigins(ImmutableList.of(Cors.Origin.of(origin)))
    .setMethods(ImmutableList.of(method))
    .setResponseHeaders(ImmutableList.of(responseHeader))
    .setMaxAgeSeconds(maxAgeSeconds)
    .build();
```
```

File: src/main/java/com/google/step/CorsContextListener.java (hunk 2 above)
Comment (cgregori) [reply src/main/java/com/google/step/CorsContextListener.java:15]: Maybe:
```suggestion
/***
    This servlet is a context listenter that runs at application startup. It configures
    the Cloud Storage bucket to accept CORS requests from the project URL.
***/
```

Comment (cgregori) [reply src/main/java/com/google/step/CorsContextListener.java:15]: Same comment here for other "comments" in the code (i.e. the comment above @WebServlet in Authenticate.java and other files)

File: src/main/java/com/google/step/MapImage.java [hunk 7]
```diff
@@ -27,13 +23,22 @@
...
+    private String url;
+
+    public MapImage(
+            double longitude,
+            double latitude,
+            String cityName,
+            int zoom,
+            int month,
+            int year,
+            String timeStamp) {
```
Comment (cgregori) [inline src/main/java/com/google/step/MapImage.java:41]: nit: I think the linter might have added some tab issues here.

File: src/main/java/com/google/step/QueryCloud.java [hunk 8]
```diff
@@ -51,20 +45,28 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+            mapImages = gson.fromJson(reader, new TypeToken<ArrayList<MapImage>>() {}.getType());
             Storage storage = StorageOptions.getDefaultInstance().getService();
-            mapImages.forEach(image -> {
-                BlobInfo blobInfo = BlobInfo.newBuilder(BUCKET_NAME, image.getObjectID()).build();
-                String url = storage.signUrl(blobInfo, 10, TimeUnit.MINUTES, Storage.SignUrlOption.withV4Signature()).toString();
-                image.setURL(url);
-            });
-        }
-        catch(SigningException e) {
+            mapImages.forEach(
```
Comment (cgregori) [inline src/main/java/com/google/step/QueryCloud.java:55]: nit: tab issues

File: src/main/java/com/google/step/SaveDrive.java [hunk 9]
```diff
@@ -143,80 +134,101 @@ private String getParentFolderID(MapImage mapImage) throws IOException {
...
-            .setSpaces("drive")
-            .setDriveId(DRIVE_ID)
-            .setIncludeItemsFromAllDrives(true) // Required parameter for querying shared drives
-            .setCorpora("drive")                // Required parameter for querying shared drives
-            .setSupportsAllDrives(true)         // Required parameter for querying shared drives
-            .setFields("files(id, name)")
-            .execute();
+        result =
+                drive.files()
+                        .list()
```
Comment (cgregori) [inline src/main/java/com/google/step/SaveDrive.java:205]: nit: Personally I like the previous format better, but it's not a huge deal

File: src/main/webapp/js/FileSaver.js [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+    }
+  } // Detect WebView inside a native macOS app by ruling out all browsers
+  // We just need to check for 'Safari' because all other browsers (besides Firefox) include that too
+  // https://www.whatismybrowser.com/guides/the-latest-user-agent/macos
+
+
+  var isMacOSWebView = /Macintosh/.test(navigator.userAgent) && /AppleWebKit/.test(navigator.userAgent) && !/Safari/.test(navigator.userAgent);
+  var saveAs = _global.saveAs || ( // probably in some web worker
+  typeof window !== 'object' || window !== _global ? function saveAs() {}
+  /* noop */
```
Comment (cgregori) [inline src/main/webapp/js/FileSaver.js:94]: what does this noop do?

File: src/main/webapp/js/FileSaver.js [hunk 11]
```diff
@@ -0,0 +1,188 @@
...
+      setTimeout(function () {
+        URL.revokeObjectURL(a.href);
+      }, 4E4); // 40s
+
+      setTimeout(function () {
+        click(a);
+      }, 0);
+    }
+  } // Use msSaveOrOpenBlob as a second approach
+  : 'msSaveOrOpenBlob' in navigator ? function saveAs(blob, name, opts) {
```
Comment (cgregori) [inline src/main/webapp/js/FileSaver.js:126]: is the colon here part of a ternerary operator or a label for a statement?

File: src/main/webapp/js/FileSaver.js [hunk 12]
```diff
@@ -0,0 +1,188 @@
...
+        a.target = '_blank';
+        setTimeout(function () {
+          click(a);
+        });
+      }
+    } else {
+      navigator.msSaveOrOpenBlob(bom(blob, opts), name);
+    }
+  } // Fallback to using FileReader and a popup
+  : function saveAs(blob, name, opts, popup) {
```
Comment (cgregori) [inline src/main/webapp/js/FileSaver.js:144]: same question here

File: src/main/webapp/js/FileSaver.js [hunk 13]
```diff
@@ -0,0 +1,188 @@
...
+        url = isChromeIOS ? url : url.replace(/^data:[^;]*;/, 'data:attachment/file;');
+        if (popup) popup.location.href = url;else location = url;
+        popup = null; // reverse-tabnabbing #460
+      };
+
+      reader.readAsDataURL(blob);
+    } else {
+      var URL = _global.URL || _global.webkitURL;
+      var url = URL.createObjectURL(blob);
+      if (popup) popup.location = url;else location.href = url;
```
Comment (cgregori) [inline src/main/webapp/js/FileSaver.js:175]: nit: "{}" braces for if else statements

File: src/main/webapp/js/FileSaver.js (hunk 10 above)
Comment (thompsond) [reply src/main/webapp/js/FileSaver.js:94]: FileSaver.js is one of the APIs for the batch download

File: src/main/webapp/js/FileSaver.js (hunk 11 above)
Comment (thompsond) [reply src/main/webapp/js/FileSaver.js:126]: FileSaver.js is one of the APIs for the batch download

File: src/main/java/com/google/step/FrontendQueryDatastore.java (hunk 3 above)
Comment (thompsond) [reply src/main/java/com/google/step/FrontendQueryDatastore.java:4]: The formatter separates static imports from normal imports

File: src/main/webapp/js/script.js [hunk 14]
```diff
@@ -29,6 +29,19 @@ $(document).ready(function() {
     fetch('/start-save-drive');
 });
 
+
+/** Once the html elements for the images are finalized, this code will be used to download the images in a zip folder */
```
Comment (theannielin) [inline src/main/webapp/js/script.js:33]: nit: add a TODO here

**Pull request merged**